Source/Core/VideoCommon/UberShaderCommon.cpp

```diff
@@ -0,0 +1,26 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "VideoCommon/UberShaderCommon.h"
+#include "VideoCommon/VideoConfig.h"
+
+namespace UberShader
+{
+void WriteUberShaderCommonHeader(ShaderCode& out, APIType api_type)
+{
+  // ==============================================
+  //  BitfieldExtract for APIs which don't have it
+  // ==============================================
+  if (!g_ActiveConfig.backend_info.bSupportsBitfield)
+  {
+    out.Write("uint bitfieldExtract(uint val, int off, int size) {\n"
```

Comment:
Mesa 17.x has a bug where it doesn't allow you to create a function named "bitfieldExtract" on OpenGL ES 3.0 even though the builtin doesn't exist until OpenGL ES 3.1. I encountered this in Android x86. I've filed a Mesa bug already.